src/app/static/src/js/components/runReport/runReport.vue

```diff
@@ -81,6 +82,32 @@
             }
         },
         methods: {
+            refreshGit: function () {
+                api.post('/git/fetch/')
+                    .then(() => {
+                        this.refreshPage()
+                    })
+                    .catch((error) => {
+                        this.error = error;
+                        this.defaultMessage = "An error occurred refreshing Git commits";
+                    });
+            },
+            refreshPage: function () {
```

Comment:
I moved all the mount logic to this function so it could be called again to 'refresh' the page after the `refreshGit` endpoint returned. This might be overdoing it though, so let me know if you think running all this logic again is unnecessary or detrimental.

Reply:
Yep, the refresh logic on this page was already a bit tangled! So a `fetch` could (in the worst case) actually remove branches/commits/reports, so we probably have to reset `gitCommits`, `reports`, `selectedBranch`, `selectedCommitId` and `selectedReport`. I think you do this indirectly, but it's not obvious. My feeling is that we should copy `gitBranches` from props to data, update it after doing git fetch, and have a watcher on it that resets the fields mentioned above. But perhaps @EmmaLRussell has another opinion on this.

Reply:
Yes, the prop should probably be renamed 'initialGitBranches' and copied into gitBranches in data which is then updated on git fetch. Arguably you could take out the prop and always fetch branches on mount as well as on git refresh to avoid this - but this would make load more sluggish. 
Yes, a watch to see when the git branches have changed in order to reset the other properties. It would be quite to retain properties which don't need to change, but that would be too much of a faff.